nuxeo-services/login/nuxeo-platform-login-cas2/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/cas2/SecurityExceptionHandler.java

```diff
@@ -66,9 +67,12 @@ public void handleException(HttpServletRequest request, HttpServletResponse resp
         }
 
         Principal principal = request.getUserPrincipal();
-        NuxeoPrincipal nuxeoPrincipal;
+        if (principal == null) {
+            LoginContext loginContext = (LoginContext) request.getAttribute("org.nuxeo.ecm.login.context");
+            principal = (Principal) loginContext.getSubject().getPrincipals().toArray()[0];
```

Comment:
This should be made a bit more robust to avoid NPE in case there's no subject or no principals.